pf/internal/pfutils/block.go

```diff
@@ -86,13 +87,18 @@ func detectSizeConstraints(x BlockLike) (int64, int64, bool) {
 	if listBlock, isList := x.(hasListValidators); isList {
 		for _, v := range listBlock.ListValidators() {
 			desc := v.Description(ctx)
-			if m := listSizeRegExp.FindStringSubmatch(desc); m != nil {
+			if m := listSizeRegExpAtLeastAtMost.FindStringSubmatch(desc); m != nil {
```

Comment:
It's unfortunate that this can't be a type case.
https://github.com/hashicorp/terraform-plugin-framework-validators/blob/48349f4e12f4283dc442114e01029baf933dfee5/listvalidator/size_at_least.go#L14-L17

Reply:
reflection could work.